lib/samson/hooks.rb

```diff
@@ -63,6 +110,32 @@ def render_views(name, view, *args)
         nil
       end
 
+      def plugin_setup
+        Samson::Hooks.plugins.
+          each(&:require).
+          each(&:precompile_assets).
+          each(&:add_lib_path).
+          each(&:add_decorators)
+      end
+
+      def render_javascripts(view)
+        Samson::Hooks.plugins.map(&:javascripts).flatten.each do |script|
+          view.instance_exec do
+            concat view.javascript_include_tag(view.asset_path(script))
```

Comment:
Actually they aren't name spaced. This will generate the URLs like ```samson.com/assets/flowdock_plugin.js``` which are of course non name spaced. I can solve this by putting the javascript inside a ```plugins``` directory. 
I don't know (apart from performance issues) what would be the advantage of compiling everything into a single application.js, the conflicts would still occur, since variables would redefine themselves (and we are not even handling version conflicts). 

I think a definite solution for this problem (asset conflicts (with names and versioning) will involve a lot more work than it is done here. I believe that sandboxing each plugin view in an IFrame with just the core scripts and their specific ones is the way to go.

This way each plugin would live on its own world, and we would handle the different javascript versions without clashes and name conflicts. 

But I really would like to have @grosser input here.